### DISCUSSION TITLE
Olimex ESP32-POE with CMT2300A

### DISCUSSION THREAD

**Discussion by letmoc** [tbnobody/OpenDTU#873] [General]
Hello, does the combination work: Olimex ESP32-POE with CMT2300A ? 
If so, which device profile would be correct or how must it be adjusted.

**Comment by DPO99** [comment]
Should work is on the whitelist from opendtu.

Hm which config do you need? Only CMT+ETH? CMT+NRF+ETH? CMT+ETH? Display? You can build it with these [informations](https://github.com/tbnobody/OpenDTU/blob/master/docs/DeviceProfiles.md).

Maybe this is good for you?

```
[
    {
        "name": "Olimex ESP32-POE CMT+ETH",
        "cmt": {
            "clk": 14,
            "cs": 5,
            "fcs": ?,
            "sdio": ?,
            "gpio2": -1,
            "gpio3": -1
        },
        "eth": {
            "enabled": true,
            "phy_addr": 0,
            "power": 12,
            "mdc": 23,
            "mdio": 18,
            "type": 0,
            "clk_mode": 3
        }
    }
]
```

EDIT_ Pins changed - Not 100% sure

**Comment by lexxman123** [comment]
Hi there, was anyone successful with connecting Olimex ESP32-POE with CMT2300A? 
I tried some combinations setting clk, cs, fcs, sdio to different GPIO pins that are exposed by ESP32-POE, but was not successful, chip status is always "not connected".

**Reply by lexxman123** [reply]
Update:
The following configuration works for me:

[
    {
        "name": "Olimex ESP32-POE CRMT300A",
        "cmt": {
            "clk": 15,
            "cs": 16,
            "fcs": 13,
            "sdio": 2,
            "gpio2": -1,
            "gpio3": -1
        },
        "eth": {
            "enabled": true,
            "phy_addr": 0,
            "power": 12,
            "mdc": 23,
            "mdio": 18,
            "type": 0,
            "clk_mode": 3
        }
    }
]

One additional obstacle when using the E49-900MBL-01 Testing Board:
I got (so it seems) a different board revision which does not match the PIN layout which is documented in the web, e.g. the PIN labeled "reset" was actually CS.
I found the correct mapping by using a circuit tester / continuity tester to check which PINs from the CMT2300A chip are connected to which PINs on the board. 
I used this specification for the CMT2300A:
![grafik](https://github.com/tbnobody/OpenDTU/assets/77163863/c902ad42-3a32-4f7b-bb59-4accbfed21a3)
![grafik](https://github.com/tbnobody/OpenDTU/assets/77163863/d36f2ad6-f8c8-4ff3-a5f3-8e7a9e8ac87d)
